README.md

```diff
@@ -416,6 +416,38 @@ server.addHandler( new FileFallbackHandler(_fs, "/jquery/jq1.11.1.js" , "/jquery
 server.addHandler( new FileFallbackHandler(_fs, "/jquery/jqm1.4.5.js" , "/jquery/jqm1.4.5.js" ,   "http://code.jquery.com/mobile/1.4.5/jquery.mobile-1.4.5.min.js",  "max-age=86400"));
 ```
 
+### Serving static files from SPIFFS
+Serve a specific file or attach a folder in SPIFFS to a root url.
+An optional Cache-Control header can be added to responses.
+An optional Last-Modified header can be added, if the request contains If-Modified-Since header with the same value the server will respond with 304 code instead of serving the file.
+```cpp
+
+// Serve static file example
+server.serveStatic("/url/file.htm", SPIFFS, "/dir/file.htm");
+// Serve static folder example
+server.serveStatic("/root-url", SPIFFS, "/root-dir");
+// Specify cache control
+server.serveStatic("/root-url", SPIFFS, "/root-dir", "max-age=86400");
+// Specify last modified (should be updated each time files are updated)
+server.serveStatic("/root-url", SPIFFS, "/root-dir", NULL, "Wed, 08 Jun 2016 22:00:00 GMT");
+
+```
+
+### HTTP Redirect (302)
+Return HTTP 302 response to redirect the client to a different location.
+An optional IPAddress can specify redirection exclusions, for example, exclude redirection to requests to the AP address.
+```cpp
+// Redirect all requests
+server.redirect("/root/file.ext", "/other-location/other-file.ext");
+// Redirect requests not for the AP address
```

Comment:
oh oh, I think I'm starting to grasp the idea :) I think I missed the part where you let another handler take care of the request.
As for a different index or any other page, I think the FileFallbackHandler needs to be updated to handle two files, like:
```cpp
server.addHandler(new FileFallbackHandler(SPIFFS, "/" , "/index-ap.html" , "/index-sta.html", "max-age=86400"));
```
the other way is to actually implement some sort of rewrite functionality, that can rewrite the request's url and host based on the request (before the handlers are checked if they can handle it)